src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -27,95 +37,172 @@ export class EditorComponent implements OnInit {
-  public getMaskData(): ImageData {
-    return this.maskData;
+  /**  Returns black transparent ImageData for single mask image. */
+  public getMaskImageData(): ImageData {
+    return this.maskImageData;
   }
 
-  /** Returns the scaled images data for reference printing scaled image after mask. */
+  /**  Returns the scaled images data for reference printing scaled image after mask. */
   public getScaledData(): ImageData {
     return this.ctx.getImageData(0, 0, this.image.width * this.scaleFactor, this.image.height * this.scaleFactor);
   }
+
+  /** 
+   *  Clears hiddeCtx to get mask Image as url to store and redraws
+   *    original image for possibility user makes more edits. 
+   *  getOriginalImageData uses hiddenCtx to pass the image data for drawing
+   *    in the mask Directive. Conflict if getOriginalImageData is called while saving the mask.
+   * @return Url for the mask image to be stored in blobstore.  
+   */
+  async getMaskBlob(): Promise<void> {
+    //  Clear canvas and put mask data to return mask as Image, 
+    this.hiddenCtx.clearRect(0,0, this.hiddenCanvas.nativeElement.width, this.hiddenCanvas.nativeElement.height);
+    this.hiddenCtx.putImageData(this.maskImageData, 0, 0);
+
+    let maskUrl = this.hiddenCanvas.nativeElement.toDataURL();
+    this.blobMask = await fetch(maskUrl).then(response => response.blob());
+    this.blobMask.lastModifiedDate = new Date();
+    this.blobMask.name = this.parentName + 'Mask.png';
+
+    //  Redraw original image if user adds more to mask
+    this.hiddenCtx.clearRect(0,0, this.hiddenCanvas.nativeElement.width, this.hiddenCanvas.nativeElement.height);
+    this.hiddenCtx.drawImage(this.image, 0, 0);
+  }
+
+  /** 
+   *  Initializes Form group and data as new
+   *  Initializes @param projectId
+   */
+  private initMaskForm() {
+    this.uploadMaskForm = new FormGroup({
+      maskName: new FormControl(),
+      labels: new FormControl(),
+    });
+
+    this.formData = new FormData();
+  }
+
+ /** 
+  *  Builds ImageBlob to be appended to form and posted.
+  *  Constructor is built in the following way: 
+  *    projectIdIn: string, 
+  *    imageNameIn: string, 
+  *    modeIn: string, 
+  *    imageIn: Blob = '',
+  *    parentImageNameIn: string = '',
+  *    newImageNameIn: string = '',
+  *    tagsIn: string = '',
+  *    deleteIn: string = 'delete'
+  *    In the case of building a mask, there is no newImageName 
+  *      but there could be tags, must pass an empty parameter for
+  *      newImageNameIn. 
+  */
+  async onSubmit(): Promise<void> {
+    // Name is a required input. If it's null, do nothing.
+    if (!this.uploadMaskForm.get('maskName').value) {
+      return;
+    }
+    
+    await this.getMaskBlob();
+
+    let imageBlob = new ImageBlob(
+      this.projectId, 
+      this.uploadMaskForm.get('maskName').value,
+      'create', 
+      this.blobMask,
+      this.parentName,
+      '', '',
+      this.uploadMaskForm.get('labels').value
+    );
```

Comment:
I understand the reasoning why you are adding the function comment with the ImageBlob.  It is hard to remember the order of the parameters of the constructor.  Instead of duplicating the comments for every function that calls ImageBlob it is better to make sure the ImageBlob calls are self-documented. This is because they are less likely to go stale and out of date.  So for most things you are using good variable names.  Where you aren't then I would add a comment that documents them.  So something like:

```
    let imageBlob = new ImageBlob(
      this.projectId, 
      /*imageName=*/this.uploadMaskForm.get('maskName').value,
      /*mode=*/'create', 
      this.blobMask,
      this.parentName,
      /*newImageName=*/'',
      /*tags=*/ '',
      /*delete=*/this.uploadMaskForm.get('labels').value
    );
```